src/models/enums.rs

```diff
@@ -95,3 +96,35 @@ impl From<&near_indexer::near_primitives::account::AccessKeyPermission> for Acce
         }
     }
 }
+
+#[derive(Debug, DbEnum, Clone)]
+#[DbValueStyle = "SCREAMING_SNAKE_CASE"]
+#[DieselType = "State_change_reason_kind"]
+#[PgType = "state_change_reason_kind"]
+pub enum StateChangeReasonKind {
+    TransactionProcessing,
+    ActionReceiptProcessingStarted,
+    ActionReceiptGasReward,
+    ReceiptProcessing,
+    PostponedReceipt,
+    UpdatedDelayedReceipts,
+    ValidatorAccountsUpdate,
+}
+
+impl From<&near_indexer::near_primitives::views::StateChangeCauseView> for StateChangeReasonKind {
+    fn from(
+        state_change_cause_view: &near_indexer::near_primitives::views::StateChangeCauseView,
+    ) -> Self {
+        match state_change_cause_view {
+            near_indexer::near_primitives::views::StateChangeCauseView::TransactionProcessing { .. } => Self::TransactionProcessing,
+            near_indexer::near_primitives::views::StateChangeCauseView::ActionReceiptProcessingStarted { .. } => Self::ActionReceiptProcessingStarted,
+            near_indexer::near_primitives::views::StateChangeCauseView::ActionReceiptGasReward { .. } => Self::ActionReceiptGasReward,
+            near_indexer::near_primitives::views::StateChangeCauseView::ReceiptProcessing { .. } => Self::ReceiptProcessing,
+            near_indexer::near_primitives::views::StateChangeCauseView::PostponedReceipt { .. } => Self::PostponedReceipt,
+            near_indexer::near_primitives::views::StateChangeCauseView::UpdatedDelayedReceipts { .. } => Self::UpdatedDelayedReceipts,
+            near_indexer::near_primitives::views::StateChangeCauseView::ValidatorAccountsUpdate { .. } => Self::ValidatorAccountsUpdate,
+            StateChangeCauseView::NotWritableToDisk => { panic!("Not expected to get NotWritableToDisk here") }
```

Comment:
Let's use tagged matching to avoid duplication of panics:

```
unexpected_state_change_cause @ NotWitableToDisk | InitialState => panic!("Unexpected state change cause {:?}", unexpected_state_change_view)
```